tools/releases/version.sh

```diff
@@ -1,30 +1,51 @@
 #!/usr/bin/env bash
 
-# Kuma version is built as follows:
+# Kuma version info has the following format:
+# version git-tag git-commit build-date envoy-version
+
+# version is built as follows:
 # 1) If a git tag is present on the current commit, then the version is a git tag (without a starting v)
 # 2) If the branch is "release-X.Y" look at the existing tags and use either X.Y.0-<shortHash> if there's none or
 #     the latest tag with a patch version increased by one (.e.g if latest tag is X.Y.1 the version will be `X.Y.2-<shortHash>`)
 # 3) In non release branch use `0.0.0-$shortHash`
 
+# git-tag - if the current HEAD has a tag then use it otherwise it's "not-tagged"
+# git-commit - HEAD sha
+# build-date - local date if built on CI
+
 # Note: this format must be changed carefully, other scripts depend on it
-exactTag=$(git describe --exact-match --tags 2> /dev/null)
+shortHash=$(git rev-parse --short HEAD 2> /dev/null || echo "no-commit")
+currentBranch=$(git rev-parse --abbrev-ref HEAD 2> /dev/null || echo "no-branch")
+envoyVersion=$("${BASH_SOURCE%/*}"/../envoy/version.sh)
+
+if [[ -z "${CI}" ]]; then
+versionDate="local-build"
+describedTag="local-build"
+longHash="local-build"
+else
+versionDate=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+describedTag=$(git describe --tags 2>/dev/null || echo "unknown")
```

Comment:
`git describe --tags` - this actually makes sense for non-tagged commits. It will - "If the tag points to the commit, then only the tag is shown. Otherwise, it suffixes the tag name with the number of additional commits on top of the tagged object
       and the abbreviated object name of the most recent commit. The result is a "human-readable" object name which can also be used to identify the commit to other git commands." and it is used [here](https://github.com/kumahq/kuma/blob/be78fca1e78d0195b7d3b74e94c892fb4011ddca/mk/kind.mk#L100) so that's why I kept it as it is. 

So the question is - should it be the same as in `k3d` ([here](https://github.com/kumahq/kuma/blob/0c7f76f695f7c95fc0576ba8e179132c614698ac/mk/k3d.mk#L138)) and I can remove the whole `describedTag` thing, or is it somehow needed for `kind` to work?

Reply:
Ok, I manually checked that `--set global.image.tag="$(BUILD_INFO_GIT_TAG)"` does not work and image pulls fail with 404. I'll remove this.